es/src/main/scala/monix/connect/es/ElasticsearchSource.scala

```diff
@@ -0,0 +1,95 @@
+package monix.connect.es
+
+import com.sksamuel.elastic4s.requests.searches.{SearchHit, SearchRequest, SearchResponse}
+import com.sksamuel.elastic4s.{ElasticClient, RequestFailure, RequestSuccess, Response}
+import monix.eval.Task
+import monix.execution.internal.InternalApi
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.collection.mutable
+
+/**
+  *  A pre-built [[monix.reactive.Observable]] implementation that
+  *  publishes documents using an elasticsearch scroll cursor.
+  *
+  * @param request [[SearchRequest]]
+  * @param client an implicit instance of a [[ElasticClient]]
+  */
+@InternalApi private[es] class ElasticsearchSource(request: SearchRequest)(implicit client: ElasticClient)
+  extends Observable[SearchHit] {
+
+  import com.sksamuel.elastic4s.ElasticDsl._
+
+  private var scrollId: String = _
```

Comment:
could that be an `Option[String]` monad that initially is set to `None`?

Reply:
Sure